addons/account/tests/test_account_payment_register.py

```diff
@@ -1509,3 +1509,50 @@ def test_group_payment_method_with_and_without_discount_and_refund(self):
         })
 
         self.assertEqual(wizard.amount, 39.50)
+
+    def test_group_payment_method_with_branch(self):
+        # create a new branch
+        self.env.company.write({
+            'child_ids': [
+                Command.create({'name': 'Branch A'}),
+                Command.create({'name': 'Branch B'}),
+            ],
+        })
+        self.cr.precommit.run()  # load the CoA
+
+        # create an invoice on the new branch
+        branch_invoices = self.env['account.move']
+        for branch in self.env.company.child_ids:
+            self.env["account.journal"].create({
+                'code': 'TEST',
+                'company_id': branch.id,
+                'name': f'{branch.name} journal',
+                'type': 'bank',
+            })
+            branch_invoices |= self.init_invoice('out_invoice', products=self.product_a, company=branch)
```

Comment:
Also make sure that the receivable account (`self.partner_a.property_account_receivable_id`) for the branch is owned by the branch as well